kotti-tests-kovenant/src/main/kotlin/com/github/adeynack/kotti/tests/kovenant/PromiseAssertions.kt

```diff
@@ -0,0 +1,44 @@
+package com.github.adeynack.kotti.tests.kovenant
+
+import com.github.adeynack.kotti.tests.AssertionUtilities.fail
+import com.github.adeynack.kotti.tests.AssertionUtilities.isA
+import nl.komponents.kovenant.Promise
+import org.amshove.kluent.ExceptionResult
+import kotlin.reflect.KClass
+
+object PromiseAssertions {
+
+    infix fun <V, E : Exception> Promise<V, E>.shouldFailWith(expectedFailureType: KClass<out E>): ExceptionResult<E> {
```

Comment:
Because then you cannot use it as an `infix` :( and I was trying to stick to the "style" of assertion that you have in [Kluent](https://github.com/MarkusAmshove/Kluent/blob/c60915862bb427f9a9b3aea0f53a871394ce3db5/docs/Exceptions.md#L20).

I however like the idea. I do believe that any function having a `clazz: Class<T>` or `KClass<T>` parameter should have an extension method to avoid providing it with the ugly `::class` thingny.